bugfix(md-icon): fix alignment issues

HitomiTenshi opened the pull request:
## Description

This PR fixes alignment issues when using smaller icon size classes (e.g. `.material-icons.md-18`) than the default one (`.material-icons.md-24`).

### What's included?

- Changes to `width` and `height` in `src/platform/core/common/styles/font/_font.scss`
- Added icon size section to `src/app/components/components/material-components/material-components.component.html`

#### Test Steps

Using `<md-icon class="md-18">filter_none</md-icon>` in an icon-button without this fix:

![electron_2017-01-11_12-14-34](https://cloud.githubusercontent.com/assets/7304212/21846826/2831d3dc-d7f9-11e6-9647-2090aeb212d5.png)

Using `<md-icon class="md-18">filter_none</md-icon>` in an icon-button **with** this fix:

![electron_2017-01-11_12-16-50](https://cloud.githubusercontent.com/assets/7304212/21846843/3c523938-d7f9-11e6-87f0-df54594c8003.png)

emoralesb05 commented:
Awesome! We will test this and merge afterwards. Thanks :tada:

HitomiTenshi commented:
Oh sorry, I didn't compile and check after doing the changes. My bad.

You guys are already using the `.md-icon-font-xl` classes on those icons, but you do not have any CSS defined for them.

I'll fix that.

HitomiTenshi commented:
I put the width and height changes into ´src/platform/core/common/styles/_icons.scss´, since `.md-icon-font-xl` is a class specific to covalent and not the official Material Icon CSS.

@indyplanets I went through the site this time and everything seems to be in order now. 👍

indyplanets commented:
@HitomiTenshi Thanks for those changes! 👍 

I've been testing this out and making sure we cover everything here. Looks like this change has some unwanted consequences. See screenshot:
![md-icon](https://cloud.githubusercontent.com/assets/104025/21910744/ff62f748-d8e2-11e6-9e35-a337552f4e98.jpg)

Might be a better option to add specific `width` and `height` styles for each `md-18`, `md-24`, etc... classes.

HitomiTenshi commented:
Alright, I have been going at this all wrong. While I was searching for the cause of this dilemma, I noticed that the icons still received `width` and `height` from somewhere. That's when I found [material 2's icon definitions](https://github.com/angular/material2/blob/master/src/lib/icon/icon.scss).

The material 2 devs add width and height definitions to `md-icon`, which seems to cause the misalignment issues on icons that have a font-size smaller than 24px.

The solution to that was simple: Replace the current definition of `1em` with `auto` on `width` and `height`, which will override material 2's `md-icon` definitions and automatically apply correct sizes when using smaller icons.

With `auto` your `md-card-colored` icons resize and align appropriately as well without the need of extra css.

I went ahead and added an Icon Size section to the docs, so you can see the effect. I also fixed a few alignment issues there while I was at it. 😎

kyleledbetter commented on src/app/components/components/material-components/material-components.component.html:
```diff
@@ -205,85 +205,121 @@
-        </button>
-        <button md-fab color="accent">
+          </button>
+          <button md-fab color="accent" style="margin-left: 10px;">
```
these inline styles are just for demo spacing right? we strive (but don't always achieve for the demos to match the code example, and we obviously wouldn't want to recommend inline styles. maybe put these buttons in a md-toolbar, or a `<div layout="row" layout-align="space-between center">` for the demo so they have proper spacing?

HitomiTenshi replied:
Well the dividers had inline styles and the radio button demo also had `style="padding-right:10px;"`, so I thought that you guys were okay with it, since it's a demo and all.

I could change the `layout-align` to `"space-between center"` if you want, but since pretty much everything else on that particular page is kinda inline styled, the page would need a re-edit someday anyway.

emoralesb05 replied:
For the most part the `material components` demo docs need to be divided so its not `1` big component with proper layouts. 

For this PR in particular, i would not make too much emphasis on inline styling since we do it everywhere else in that view, and just focus on the fix and display of the different fonts (which its there now)

indyplanets commented:
@HitomiTenshi 
<img width="438" alt="screen shot 2017-01-13 at 3 43 12 pm" src="https://cloud.githubusercontent.com/assets/104025/21946754/75cb3ee4-d9a7-11e6-9ce0-f099cfed9256.png">
Thanks for adding that section! 👍 🦄 
PR approved for merge.

kyleledbetter commented:
thx for the fix @HitomiTenshi! 🍺
